satpy/tests/test_readers.py

```diff
@@ -510,7 +521,15 @@ def test_no_parameters(self, viirs_file):
         assert list(ri.keys()) == ['viirs_sdr']
         assert ri['viirs_sdr'] == [viirs_file]
 
-    def test_bad_sensor(self, viirs_file):
+    def test_no_parameters_atms(self, atms_file):
+        """Test with no limiting parameters."""
+        from satpy.readers import find_files_and_readers
+
+        ri = find_files_and_readers()
+        assert list(ri.keys()) == ['atms_sdr_hdf5']
+        assert ri['atms_sdr_hdf5'] == [atms_file]
```

Comment:
I don't think this is needed, because it's basically the same think as the function above (or that could be parametrized).
But maybe it would be important to test what `find_files_and_readers` returns when both a viirs file and an atms file are in the same directory?

Reply:
Indeed, yes! I just tested this and the `find_files_and_readers` seem to never finish! :-(

Reply:
No, my bad. It seemed so, but it actually does terminate and gives a nice (expected) output.